### PR TITLE
Review Commit

### PR DESCRIPTION
**Pull request by canon20paul**
## Does my coding partner follow JS best practices? Task

### CODE REVIEW

**Comment by laiifuu** [comment]
Hi @canon20paul 👋

Good job so far on your project. There are still some issues you'll need to work on though to make your project even better. 
- You code is not indented as it should be, please fix the indentation to have 2 spaces across your JavaScript code and HTML code in the src file. 

Please Check the comments under the review for more details.

File: src/TasklistFile.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { update } from "lodash";
```
Comment (laiifuu) [inline src/TasklistFile.js:1]: The import from the lodash package here is unnecessary, so please remove it.

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+       <title>To do List</title>
+</head>
+<body>
+      <!-- header section-->
+  <header class="navigation">
+     </header>
```
Comment (laiifuu) [inline src/index.html:12]: The header here is unnecessary for this project, so please remove it.

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        <main>
+            <div action="" class="frm" id="frm">
+                <h1 class="logo">To do List</h1>
+                <label for="addTT"></label> <div class="addDiv" title="Enter to Save"><input type="text" id="addTT" class="addT" placeholder="Add to your list..." title="Press Enter to Save" required><div class="save" title="Press Enter to Save">&#9851;</div></div>
+           <div id="task-list">
+           </div>
+            <button type="submit" id="btnDelComplete" >Clear all completed</button>
+        </div>
+        </main>
+        <footer></footer>
```
Comment (laiifuu) [inline src/index.html:22]: The footer is also unnecessary, so please remove it.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import _, { indexOf } from 'lodash';
+import './index.css';
+import {TaskList} from './TasklistFile.js';
+import {Interaction} from './interactive_List.js'
+const textInputAdd = document.getElementById('addTT');
+const taskList = document.getElementById('task-list');
+function check(status) {
```
Comment (laiifuu) [inline src/index.js:7]: Adding spacing here between that last `const` declaration and the function could be helpful for readability.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import _, { indexOf } from 'lodash';
```
Comment (laiifuu) [inline src/index.js:1]: The lodash package is unnecessary for this project, so please remove this `import`.

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  task.addEventListener('dblclick', (e) => {
+    task.contentEditable=true;
+  });
+  task.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') { task.contentEditable = false;
+      for(var i=0; i < TaskList.taskArray.length; i=1+i){
+                if(TaskList.taskArray[i].index==task.id){
+    TaskList.taskArray[i].description=e.target.innerHTML;
+        }
+      }
```
Comment (laiifuu) [inline src/index.js:54]: Please avoid using this type of `for` loop, and use `forEach()` instead, to respect the KISS principle.

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  if (status === true) { return "checked";
+ } else { return "unchecked";
+ }}
+function styleTask(status) { if (status === true) { 
+  return "task taskStyle" } else { return "task" 
+  }  }
+function display() {
+  TaskList.taskArray = JSON.parse(localStorage.getItem('coward')) || [];
+  taskList.innerHTML = ''
+ TaskList.taskArray.forEach((task, i) => {
```
Comment (laiifuu) [inline src/index.js:17]: In this line, you declared a parameter `task`, that is never used afterwards, so please remove this parameter.

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+ taskList.innerHTML += `<div class="taskR"><input type="checkbox" ${check(TaskList.taskArray[i].completed)} 
+ class="box" value=${i}> <p id=${i} class="${styleTask(TaskList.taskArray[i].completed)}" title="Double Click to
+  Edit & Enter to Save">${TaskList.taskArray[i].description}</p><div class="img" id="deleteBtn ${i}" title="${i}"></div></div>`;
+  }  );
+}
+display();
+//Add Task Event Listner
+document.querySelector('.save').addEventListener('click', (e) => {
```
Comment (laiifuu) [inline src/index.js:25]: In this line, you introduced the parameter `e` for the function passed inside the event listener, but it is never used, so please remove this parameter.

File: src/index.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+})
+//Add Task Event Listner
+textInputAdd.addEventListener('keypress',  (e) => {
+    if (e.key === 'Enter') { TaskList.addTask(textInputAdd.value);
+   location.reload();}
+ })
+ //Delete Task Event Listner
+  const deleteBtn = document.querySelectorAll('.img');
+   deleteBtn.forEach((Btn) => {
+          Btn.addEventListener('click', (e) => {
```
Comment (laiifuu) [inline src/index.js:37]: In this line, you introduced the parameter e for the function passed inside the event listener, but it is never used, so please remove this parameter.

File: src/index.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+                TaskList.delete(Btn.title);
+         location.reload();
+    });
+  });
+//EditTask Event Listner
+const taskArray = document.querySelectorAll('.task');
+taskArray.forEach((task) => {
+  task.addEventListener('dblclick', (e) => {
```
Comment (laiifuu) [inline src/index.js:45]: In this line, you introduced the parameter e for the function passed inside the event listener, but it is never used, so please remove this parameter.

File: src/index.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        }
+      }
+      localStorage.setItem('coward', JSON.stringify(TaskList.taskArray));
+  }
+});
+});
+//  Checkbox Event Listner
+  const checkArray = document.querySelectorAll('.box');
+  checkArray.forEach((checkB,i) => {
+    checkB.addEventListener('click', (e) => {
```
Comment (laiifuu) [inline src/index.js:62]: In this line, you introduced the parameter e for the function passed inside the event listener, but it is never used, so please remove this parameter.

**Review by laiifuu — COMMENTED** [review]
Hi @canon20paul 👋 

Good job on the work you've done on your project so far :clap:. 
There are still some issues you'll need to work on though to make your project better. 

so-close


One if the issues I found was that the majority of your files are not indented as they should be. Please fix the indentation in your HTML file and JavaScript files inside your `src` directory to have 2 spaces. 

Please check the comments above for the rest of the issues. 

💻Happy coding💻  
✌️

File: src/index.js (hunk 11 above)
Comment (canon20paul) [reply src/index.js:62]: Hi Hayet. Thanks a lot. I appreciate your Efforts. it helped a lot.

**Comment by canon20paul** [comment]
Kindly your Efforts are very much appreciated

**Pull request merged**